### DISCUSSION TITLE
Using Loopback with Apache Cassandra

### DISCUSSION THREAD

**Discussion by gtamas** [loopbackio/loopback-next#7583] [Q&A] [answered]
Hi

I'm using the LB4 with latest Cassandra connector and Cassandra 3.x

It mostly works so far, but I've got a couple of questipns:

- How to store arrays? In my data model, I have this:

```
  @property.array(String, {
    default: [],
    jsonSchema: {
      uniqueItems: true,
      type: 'string',
      format: 'uri-reference',
      errorMessage: 'An array of file URLs is expected',
    },
  })
  attachments?: string[];
```

But Cassandra just creates this as simple text type.

Same thing happens if try to use any of the Cassandra types supported by loopback (uuid, timeuuid, tuple). All I get is a text field.

Is this supposed to work? 

If not, is there a better connector for LB4? The official one seems to be working, but it's kinda old. Last commit was ages ago. Is this still officially maintained by LB team / Strongloop?

Thanks in advance

**Comment by marioestradarosa** [comment, marked as answer]
@gtamas , maybe you need to hint cassandra connector about the real type like so:

```ts
  @property.array(String, {
    default: [],
     cassandra: {
            dataType: 'List<Text>'
          },
    jsonSchema: {
      uniqueItems: true,
      errorMessage: 'An array of file URLs is expected',
    },
  })
  attachments?: string[];
```

**Comment by gtamas** [comment]
thanks it worked perfectly!
